networkx/algorithms/similarity.py

```diff
@@ -748,7 +741,10 @@ def match_edges(u, v, pending_g, pending_h, Ce, matched_uv=[]):
         # only attempt to match edges after one node match has been made
         # this will stop self-edges on the first node being automatically deleted
         # even when a substitution is the better option
-        if matched_uv:
+        if matched_uv is None or len(matched_uv) == 0:
```

Comment:
I would've expected this to bork with a `TypeError` when matched_uv is None because both expressions are evaluated, but apparently that's not the case?! I didn't realize Python evaluated or's this intelligently, so I learned something new :)

Reply:
It seems that both `and` and `or` do not evaluate the second value if the first determines the result.  Nice! :}

Reply:
The beauty of Python :D